code/test/CoreTemplateStudio.Core.Test/Extensions/StringExtensionsTests.cs

```diff
@@ -10,11 +10,55 @@ namespace Microsoft.Templates.Core.Test.Extensions
     [Trait("ExecutionSet", "Minimum")]
     public class StringExtensionsTests
     {
+        [Theory]
+        [InlineData("string")]
+        [InlineData("string23TEST")]
+        public void ObfuscateSHA_ShouldNotBeEmpty(string value)
+        {
+            var actual = value.ObfuscateSHA();
+
+            Assert.NotEmpty(actual);
+        }
+
+        [Theory]
+        [InlineData("string")]
+        [InlineData("string23TEST")]
+        public void ObfuscateSHA_ShouldNotBeEqualsToTheValue(string value)
+        {
+            var actual = value.ObfuscateSHA();
+
+            Assert.False(value.Equals(actual, StringComparison.Ordinal));
+        }
+
+        [Theory]
+        [InlineData("string")]
+        [InlineData("string23TEST")]
+        public void ObfuscateSHA_ShouldNotContainLowercase(string value)
```

Comment:
This is a really curious test.
Without further information, it's not clear why the method under test should never return a lower case character.
It looks like you might be indirectly testing something tied to the implementation of the method being tested. Does the code that consumes the value returned by the method under test need to have upper case characters.

Reply:
We would change the name of the function to just be **Obfuscate** (the name came from 2017). 
Would it be "**Obfuscate_ShouldHaveBeenConvertedToUppercase**" a better name for this? As the last step of the function is converting the generated string to upper using: ToUpperInvariant.
We think that this method (Obfuscate) should probably be moved to be a private method within Microsoft.Templates.Core.Packaging.TemplatePackage as it´s the only place where it is needed.